index.c

```diff
@@ -927,9 +929,6 @@ int index_color(int line)
  */
 void mutt_draw_statusline(int cols, const char *buf, size_t buflen)
 {
-  if (!buf)
```

Comment:
This check is redundant. There is one more 10 lines below.

Reply:
I’d rather fail earlier than later

Reply:
Ah, just 10 lines and happens nothing in between.

Then what about moving both checks here?

Reply:
Definitely!

Reply:
Okay.

(I just wondered–because I see at lots of places–that are these `buf != NULL` checks really necessary? The most places I think `buf` is some array on stack, but if it is `malloc()`ed, the `NULL` check should be done immediately after allocation, so `NULL`s should never be passed to any functions.)